x/aswap/codec.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+
+package aswap;
+
+import "github.com/iov-one/weave/codec.proto";
+import "github.com/iov-one/weave/coin/codec.proto";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+// Swap is designed to hold some coins for atomic swap, locked by preimage_hash
+message Swap {
+    // metadata is used for schema versioning support
+    weave.Metadata metadata = 1;
+
+    bytes sender = 2 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+    // preimage hash is a hash of user's secret preimage that would serve as a "password"
+    // to unlock the coins in the swap should the transaction be a success
+    bytes preimage_hash = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Condition"];
```

Comment:
req: can you make this an address?

Reply:
@alpe it does not need to be an address afaik, why?

Reply:
On the other hand I think I see your point - easier to compare. Will do